synapseclient/client.py

```diff
@@ -2172,6 +2172,62 @@ def getTeamMembers(self, team):
         for result in self._GET_paginated('/teamMembers/{id}'.format(id=id_of(team))):
             yield TeamMember(**result)
 
+    def get_membership_status(self, userid, team):
```

Comment:
What if they already have an open invitation? Should you check that too?

https://docs.synapse.org/rest/GET/user/id/openRequest.html

Reply:
What we want is: https://docs.synapse.org/rest/GET/team/id/openInvitation.html.  Unfortunately we can't exactly just check for a user's open invitations because the rest call doesn't allow you to put in email addresses.